src/transaction/payment.ts

```diff
@@ -90,7 +91,22 @@ function createPaymentTransaction(address: string, paymentArgument: Payment
   applyAnyCounterpartyEncoding(payment)
 
   if (address !== payment.source.address) {
-    throw new ValidationError('address must match payment.source.address')
+    try {
+      const {
+        classicAddress,
+        tag
+      } = xAddressToClassicAddress(address)
+      if (classicAddress !== payment.source.address ||
+        (tag !== false && tag !== payment.source.tag) ||
+        (payment.source.tag !== undefined && payment.source.tag !== tag)) {
+          throw new ValidationError('address must match payment.source; if either contains a tag, it must match.')
```

Comment:
ditto here, will be caught immediately. and then swallowed. What you could do is re-throw any validation errors:

```
    } catch (err) {
      if (err instanceof ValidationError) {
        throw err;
    }
    /...
```

But it might be easier to just move this part out of the try block entirely:

```
let addr;
try {
      addr = xAddressToClassicAddress(address)
} catch { ... }
// move the rest of the logic down here, now that the addr variable is set
```